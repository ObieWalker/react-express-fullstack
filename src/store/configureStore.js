import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk'
import reduxImmutableStateInvariant from 'redux-immutable-state-invariant';
import rootReducer from '../reducers';

const enhancers = window.__REDUX_DEVTOOLS_EXTENSION__ && 
  window.__REDUX_DEVTOOLS_EXTENSION__();

const initialState = {};

export const configureStore = () => {
  // const store = createStore(
  //   rootReducer, 
  //   initialState,
  //   enhancers,
  //   applyMiddleware(thunk)
  // );
  const store = compose(
    applyMiddleware(thunk),
    window.devToolsExtension ? window.devToolsExtension() : f => f
  )(createStore)(rootReducer);
  return store;
}