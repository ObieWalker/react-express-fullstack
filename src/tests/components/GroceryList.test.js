import React from 'react';
import ReactDOM from 'react-dom';
import expect from 'expect';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import connect, {GroceryList} from '../../components/GroceryList';

const mockStore = configureStore();
let mounted;
let props;
let wrapper;
let id = 'dfsddsd'
let grocery = {
  id: 'adsfadsfads',
  name : 'randomName',
  price : 'randomPrice'
}
const errors = {
  groceryName : '',
  groceryPrice: ''
}

const onShow = true;
const addGrocery = jest.fn(() => Promise.resolve({}));
const buyGrocery = jest.fn(() => Promise.resolve({}));
const deleteGrocery = jest.fn(() => Promise.resolve({}));
const editGrocery = jest.fn(() => Promise.resolve({}));
const  groceries = [{}]

const getComponent = () => {
  if (!mounted) {
    props = {
      groceries,
      addGrocery,
      deleteGrocery,
      editGrocery,
      buyGrocery,
      onShow
    };
    mounted = shallow(<GroceryList {...props} />);
  }
  return mounted;
};

describe('GroceryList Component', () => {
  beforeEach(() => {});
  wrapper = getComponent();

  it('renders component successfully', () => {
    wrapper = getComponent();
    expect(wrapper).toMatchSnapshot();
  });

  describe('Check functions are called', () => {
    it('should have a method handleHide', () => {
      expect(wrapper.instance().handleHide).toBeDefined();
    });
    it('should have a method that updateDetails', () => {
      expect(wrapper.instance().updateDetails).toBeDefined();
    });
    it('should have a method that addGrocery', () => {
      expect(wrapper.instance().addGrocery).toBeDefined();
    });
    it('should have a method that editGrocery', () => {
      expect(wrapper.instance().editGrocery).toBeDefined();
    });
    it('should have a method that onDelete', () => {
      expect(wrapper.instance().onDelete).toBeDefined();
    });
    it('should have a method that onBuy', () => {
      expect(wrapper.instance().onBuy).toBeDefined();
    });
    it('should have a method that handleShowEdit', () => {
      expect(wrapper.instance().handleShowEdit).toBeDefined();
    });
  })

  describe('Checks state change', () => {
    it('should have a method handleHide', () => {
      wrapper = getComponent();
      wrapper.instance().handleHide();
      expect(wrapper.state().show).toEqual(false);
      expect(wrapper.state().groceryName).toEqual('');
      expect(wrapper.state().groceryPrice).toEqual('');
    });
    it('should have a method handleShowEdit', () => {
      wrapper = getComponent();
      wrapper.instance().handleShowEdit(grocery);
      expect(wrapper.state().show).toEqual(true);
      expect(wrapper.state().groceryName).toEqual('randomName');
      expect(wrapper.state().groceryPrice).toEqual('randomPrice');
    });
    it('should have a method addGrocery', () => {
      wrapper = getComponent();
      const e = { preventDefault: () => undefined };
      e.preventDefault = jest.fn();
      wrapper.instance().addGrocery(e);
      expect(e.preventDefault).toHaveBeenCalled();
      expect(wrapper.state().groceryName).toEqual('');
      expect(wrapper.state().groceryPrice).toEqual('');
    });
    it('should have a method addGrocery', () => {
      grocery = {
        name : '',
        price : ''
      }
      wrapper = getComponent();
      const e = { preventDefault: () => undefined };
      e.preventDefault = jest.fn();
      wrapper.instance().addGrocery(e);
      expect(e.preventDefault).toHaveBeenCalled();
      expect(wrapper.state().groceryName).toEqual('');
      expect(wrapper.state().groceryPrice).toEqual('');
    });
    it('should have a method editGrocery', () => {
      wrapper = getComponent();
      wrapper.instance().editGrocery(grocery);
      expect(wrapper.state().groceryName).toEqual('');
      expect(wrapper.state().groceryPrice).toEqual('');
    });
    it('should have a method editGrocery', () => {
      grocery = {
        name : '',
        price : ''
      }
      wrapper = getComponent();
      wrapper.instance().editGrocery(grocery);
      expect(wrapper.state().groceryName).toEqual('');
      expect(wrapper.state().groceryPrice).toEqual('');
    });
    it('should have a method onDelete', () => {
      wrapper = getComponent();
      wrapper.instance().onDelete(id);
    });
    it('should have a method onBuy', () => {
      wrapper = getComponent();
      wrapper.instance().onBuy(id);
    });
    it('should click on Add Grocery button', () => {
      wrapper = getComponent();
      wrapper
      .find('.btn')
      .simulate('click');
    });
  })
});

describe('Connected Component', () => {
    describe('Connected mounted', () => {
      it('component successfully rendered', () => {
        const store = mockStore({
          addGrocery,
          deleteGrocery,
          buyGrocery,
          editGrocery,
          groceries
        });
        wrapper = shallow(<connect
          store={store}
          addGrocery={addGrocery}
          deleteGrocery={deleteGrocery}
          buyGrocery={buyGrocery}
          editGrocery={editGrocery}
          groceries={groceries}
        />);
        expect(wrapper.length).toBe(1);
      });
    });
  });
