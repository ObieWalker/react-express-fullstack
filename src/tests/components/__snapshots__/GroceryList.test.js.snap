// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroceryList Component renders component successfully 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GroceryList
    addGrocery={[MockFunction]}
    buyGrocery={[MockFunction]}
    deleteGrocery={[MockFunction]}
    editGrocery={[MockFunction]}
    groceries={
      Array [
        Object {},
      ]
    }
    onShow={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Grocery Listing  
        </h1>,
        <div
          className="p-3 mb-2 bg-info text-white"
          style={
            Object {
              "float": "right",
              "padding": "2%",
            }
          }
        >
          <h4>
            <TotalPrice
              groceries={
                Array [
                  Object {},
                ]
              }
            />
          </h4>
        </div>,
        <GroceryTable
          addGrocery={[Function]}
          buyGrocery={[Function]}
          deleteGrocery={[Function]}
          groceries={
            Array [
              Object {},
            ]
          }
          onShow={[Function]}
        />,
        <GroceryModal
          addGrocery={[Function]}
          editGrocery={[Function]}
          errors={Object {}}
          grocery={Object {}}
          groceryName=""
          groceryPrice=""
          onHide={[Function]}
          show={false}
          updateDetails={[Function]}
        />,
        <button
          className="btn btn-lg btn-primary"
          onClick={[Function]}
        >
          Add Grocery
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Grocery Listing  ",
        },
        "ref": null,
        "rendered": "Grocery Listing  ",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h4>
            <TotalPrice
              groceries={
                Array [
                  Object {},
                ]
              }
            />
          </h4>,
          "className": "p-3 mb-2 bg-info text-white",
          "style": Object {
            "float": "right",
            "padding": "2%",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <TotalPrice
              groceries={
                Array [
                  Object {},
                ]
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "groceries": Array [
                Object {},
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "h4",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "addGrocery": [Function],
          "buyGrocery": [Function],
          "deleteGrocery": [Function],
          "groceries": Array [
            Object {},
          ],
          "onShow": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "addGrocery": [Function],
          "editGrocery": [Function],
          "errors": Object {},
          "grocery": Object {},
          "groceryName": "",
          "groceryPrice": "",
          "onHide": [Function],
          "show": false,
          "updateDetails": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add Grocery",
          "className": "btn btn-lg btn-primary",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Add Grocery",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Grocery Listing  
          </h1>,
          <div
            className="p-3 mb-2 bg-info text-white"
            style={
              Object {
                "float": "right",
                "padding": "2%",
              }
            }
          >
            <h4>
              <TotalPrice
                groceries={
                  Array [
                    Object {},
                  ]
                }
              />
            </h4>
          </div>,
          <GroceryTable
            addGrocery={[Function]}
            buyGrocery={[Function]}
            deleteGrocery={[Function]}
            groceries={
              Array [
                Object {},
              ]
            }
            onShow={[Function]}
          />,
          <GroceryModal
            addGrocery={[Function]}
            editGrocery={[Function]}
            errors={Object {}}
            grocery={Object {}}
            groceryName=""
            groceryPrice=""
            onHide={[Function]}
            show={false}
            updateDetails={[Function]}
          />,
          <button
            className="btn btn-lg btn-primary"
            onClick={[Function]}
          >
            Add Grocery
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Grocery Listing  ",
          },
          "ref": null,
          "rendered": "Grocery Listing  ",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h4>
              <TotalPrice
                groceries={
                  Array [
                    Object {},
                  ]
                }
              />
            </h4>,
            "className": "p-3 mb-2 bg-info text-white",
            "style": Object {
              "float": "right",
              "padding": "2%",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <TotalPrice
                groceries={
                  Array [
                    Object {},
                  ]
                }
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "groceries": Array [
                  Object {},
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "h4",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "addGrocery": [Function],
            "buyGrocery": [Function],
            "deleteGrocery": [Function],
            "groceries": Array [
              Object {},
            ],
            "onShow": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "addGrocery": [Function],
            "editGrocery": [Function],
            "errors": Object {},
            "grocery": Object {},
            "groceryName": "",
            "groceryPrice": "",
            "onHide": [Function],
            "show": false,
            "updateDetails": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add Grocery",
            "className": "btn btn-lg btn-primary",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add Grocery",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
